x-pack/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -199,6 +208,21 @@ export const signalRulesAlertType = ({
             result.bulkCreateTimes.push(bulkCreateDuration);
           }
         } else {
+          let listClient;
+          if (
+            process.env.ELASTIC_XPACK_SIEM_EXCEPTIONS_LISTS &&
+            process.env.ELASTIC_XPACK_SIEM_EXCEPTIONS_LISTS === 'true'
+          ) {
```

Comment:
Is it possible to use this so we don't introduce another environment variable for lists?

```ts
          let listClient: ListClient | undefined;
          if (hasListsFeature()) {
            if (lists == null) {
              throw new Error('lists plugin unavailable during rule execution');
            }
            listClient = lists.getListClient(
              services.callCluster,
              spaceId,
              updatedByUser ?? 'elastic'
            );
          }
```

Also that would type `listClient` as well. If we have to introduce a new environment variable can we just introduce it within the file:

```ts
x-pack/plugins/siem/server/lib/detection_engine/feature_flags.ts
```

And label it as `ExceptionLists`, but I would rather prefer for us to keep our environment variables to a minimum for feature flags.

If you added this because of the way the end to end integration tests are running and you need the lists to be available and not throw then you can add this line:

```ts
'--xpack.lists.enabled=true',
```

to the file:

```ts
kibana/x-pack/test/detection_engine_api_integration/common/config.ts
```

Like so:

```ts
      kbnTestServer: {
        ...xPackApiIntegrationTestsConfig.get('kbnTestServer'),
        serverArgs: [
          ...xPackApiIntegrationTestsConfig.get('kbnTestServer.serverArgs'),
          `--xpack.actions.whitelistedHosts=${JSON.stringify([
            'localhost',
            'some.non.existent.com',
          ])}`,
          `--xpack.actions.enabledActionTypes=${JSON.stringify(enabledActionTypes)}`,
          '--xpack.eventLog.logEntries=true',
          '--xpack.lists.enabled=true',
          ...disabledPlugins.map((key) => `--xpack.${key}.enabled=false`),
          `--plugin-path=${path.join(__dirname, 'fixtures', 'plugins', 'alerts')}`,
```

And it should enable and run the lists plugin along with the end to end tests.

Tested it out just now, and it looked like that would be the fix to remove extra process.env from above and it should all work with end to end tests and everything will be happy.

Reply:
Perfect - I added it mostly because of the intricacies involved with the e2e tests with the pre-existing list env var. I wasn't sure if we wanted to add the xpack variable there so I figured I'd be safer to just add another env var and see if anyone had other suggestions. I'll update the e2e config.